sklearn/cluster/k_means_.py

```diff
@@ -165,9 +165,24 @@ def _tolerance(X, tol):
     return np.mean(variances) * tol
 
 
-def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
-            n_init=10, max_iter=300, verbose=False,
-            tol=1e-4, random_state=None, copy_x=True, n_jobs=1,
+def _check_sample_weight(X, sample_weight):
+    """Set sample_weight if None, and check for correct dtype"""
+    n_samples = X.shape[0]
+    if sample_weight is None:
+        return np.ones(n_samples, dtype=X.dtype)
+    else:
+        # verify that the number of samples is equal to the number of weights
+        if n_samples != len(sample_weight):
+            raise ValueError("n_samples=%d should be == len(sample_weight)=%d"
+                             % (n_samples, len(sample_weight)))
```

Comment:
I didn't change this to `check_consistent_length`, because that would result in a very generic error message. Let me know if you'd still like me to change it.